### PR TITLE
RUMM-1528 Add billing attribute to rum events

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

In this PR we are conforming to Lite session billing plan for RUM Mobile by adding the billing property to all our RUM JSON events. Because the billing plans are declared as `Enums` with `Number` value type we had to fix the `PokoGenerator` as it was not handling this feature properly and was adding it as a `String` property to the serialized event instead of a `Primitive` property.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -6,15 +6,6 @@
           <package name="kotlinx.android.synthetic" alias="false" withSubpackages="true" />
         </value>
       </option>
-      <option name="PACKAGES_IMPORT_LAYOUT">
```
Comment (xgouchet) [inline .idea/codeStyles/Project.xml:9]: Is this change necessary ?

File: buildSrc/src/main/kotlin/com/datadog/gradle/plugin/jsonschema/PokoGenerator.kt [hunk 2]
```diff
@@ -180,16 +184,16 @@ class PokoGenerator(
...
-            PropertySpec.builder(ENUM_CONSTRUCTOR_JSON_VALUE_NAME, String::class, KModifier.PRIVATE)
+            PropertySpec.builder(ENUM_CONSTRUCTOR_JSON_VALUE_NAME, jsonValueType, KModifier.PRIVATE)
                 .initializer(ENUM_CONSTRUCTOR_JSON_VALUE_NAME)
                 .build()
         )
-
+        val parameterFormat = resolveEnumConstantFormat(definition)
         definition.values.forEach { value ->
             enumBuilder.addEnumConstant(
                 value.enumConstantName(),
```
Comment (xgouchet) [inline buildSrc/src/main/kotlin/com/datadog/gradle/plugin/jsonschema/PokoGenerator.kt:194]: Maybe we also need to change the `enumConstantName()` resolution in case the type is `Number`.  Having backticks surrounding all values makes this not user friendly

Instead of 
```kotlin
    enum class Island(
        private val jsonValue: Number
    ) {
        `4`(4),
        `8`(8),
        `15`(15),
        `16`(16),
        `23`(23),
        `42`(42);
    }
```

We could have

```kotlin
    enum class Island(
        private val jsonValue: Number
    ) {
        ISLAND_4(4),
        ISLAND_8(8),
        ISLAND_15(15),
        ISLAND_16(16),
        ISLAND_23(23),
        ISLAND_42(42);
    }
```

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (mariusc83) [reply .idea/codeStyles/Project.xml:9]: it is something that my AS always asks for ...it seems that it slept here...I would actually like to commit it if it doesn't bother anyone and avoid removing it all the time manually from the commit.

Comment (xgouchet) [reply .idea/codeStyles/Project.xml:9]: Or we could remove the `codestyle/*` from tracked files altogether. We already have ktlint to ensure we share the same formatting

File: buildSrc/src/test/kotlin/com/datadog/gradle/plugin/jsonschema/TestDefinitions.kt [hunk 3]
```diff
@@ -513,6 +513,22 @@ val Style = TypeDefinition.Class(
...
+    name = "Jacket",
+    properties = listOf(
+        TypeProperty(
+            "size",
+            TypeDefinition.Enum(
+                "Size",
+                JsonType.NUMBER,
+                listOf("1", "2", "3", "4")
+            ),
+            defaultValue = 1.0,
```
[the remark this answers is not in the document]
Comment (mariusc83) [reply buildSrc/src/test/kotlin/com/datadog/gradle/plugin/jsonschema/TestDefinitions.kt:526]: It doesn't matter actually as we always default to `Double` if it's a Number value. I will change it there also to align.

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]
